server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3090,7 +3136,7 @@ SnapshotsInProgress computeUpdatedState() {
                     changedCount,
                     startedCount
                 );
-                return updated;
+                return updated.withUpdatedNodeIdsForRemoval(initialState);
```

Comment:
Should this also be guarded by `supportsNodeRemovalTracking`?

Reply:
Ah yes good point, thanks. See 24f7e4551f2f44ffad93c6f6332cbd1974a5b8fb.